src/mlpack/methods/ne/genome.hpp

```diff
@@ -0,0 +1,334 @@
+/**
+ * @file genome.hpp
+ * @author Bang Liu
+ *
+ * Definition of the Genome class.
+ */
+#ifndef MLPACK_METHODS_NE_GENOME_HPP
+#define MLPACK_METHODS_NE_GENOME_HPP
+
+#include <cstddef>
+#include <cassert>
+#include <map>
+
+#include <mlpack/core.hpp>
+
+#include "link_gene.hpp"
+#include "neuron_gene.hpp"
+#include "utils.hpp"
+
+namespace mlpack {
+namespace ne {
+
+/**
+ * This class defines a genome.
+ # A genome is consist of a group of genes.
+ */
+class Genome {
+ public:
+  // Neurons.
+  std::vector<NeuronGene> aNeuronGenes;
+
+  // Links.
+  std::vector<LinkGene> aLinkGenes;
+
+  // Default constructor.
+  Genome() {
+    aId = -1;
+    aNumInput = 0;
+    aNumOutput = 0;
+    aFitness = DBL_MAX;
+  }
+  
+  // Parametric constructor.
+  Genome(int id,
```

Comment:
Some style issue here, we should:

- To be consistent with the rest of the codebase use the same naming conventions for the parameter and function names e.g. numInput instead of aNumInput. For more informations take a look at: https://github.com/mlpack/mlpack/wiki/DesignGuidelines#naming-conventions. Also, we tend to be very picky about coding standards and conventions, so if you find inconsistencies in existing code or things that are unclear and should be clarified, please feel free to point them out.

- We use doxygen to generate code documentation, so it would be great if you could adapt the style accordingly e.g.

```
/*
 * Construct the Genome object with the given parameters.
 *
 * @param id Genome identifier.
 * @param neuronGenes List of genes to construct the genome.
 * @param linkGenes List of links to construct the genome.
 * @param numInput Number of input genes.
 * @param numOutput Number of output genes.
 * @param fitness Initial fitness.
 */
Genome(int id,
       const std::vector<NeuronGene>& neuronGenes,
       const std::vector<LinkGene>& linkGenes,
       int numInput,
       int numOutput,
       double fitness):
```